std/algorithm/comparison.d

```diff
@@ -1215,12 +1219,15 @@ Iterates the passed arguments and return the maximum value.
 Params:
     args = The values to select the maximum from. At least two arguments must
     be passed.
+    propagateNaN = A flag determining whether to propagate or ignore $(NAN) if it
+    is encountered amongst args. Has no effect if there are no floating-point
+    values in the args.
 
 Returns:
     The maximum of the passed-in args. The type of the returned value is
     the type among the passed arguments that is able to store the largest value.
 */
-MaxType!T max(T...)(T args)
+MaxType!T max(PropagateNaN propagateNaN = Yes.propagateNaN, T...)(T args)
```

Comment:
Nitpick: 
```D
PropagateNaN propagateNaN = PropagateNaN.yes
```
Readers can see from documentation, that he can work without `std.typecons`